SimulationAgent/Agent.cs

```diff
@@ -80,6 +80,10 @@ public class Agent : ISimulationAgent
         private Thread[] devicesTelemetryThreads;
         private List<IDeviceTelemetryTask> devicesTelemetryTasks;
 
+        // The thread responsible for replaying simulations from a file
+        private Thread deviceReplayThread;
```

Comment:
It might need to be but I'm not going to worry about that until it needs to be that way.

Reply:
Multiple threads are used for telemetry so that telemetry requests can be sent in parallel. Unless this is needed for replay, we shouldn't need to use multiple threads here.